src/main/server.ts

```diff
@@ -68,21 +116,9 @@ class JupyterServer {
             this._info.url = `http://localhost:${appConfig.jlabPort}`;
             this._info.token = appConfig.token;
 
-            // note: traitlets<5.0 require fully specified arguments to
-            // be followed by equals sign without a space; this can be
-            // removed once jupyter_server requires traitlets>5.0
-            this._nbServer = execFile(this._info.environment.path, [
```

Comment:
this well-quoted approach is probably still more robust against path shenanigans. 

If you know where the conda shell is, at least _it_ will run. You can couple that with the `run` subcommand. So it coudl actually be:

```ts
[
  pathToCondaBin,
  'run', 
  '--path', pathToBaseEnv,
  '--no-capture-output',
  '--cwd', pathToStartIn,
  'jupyter-lab',
```

Reply:
Nice. Still need some if/else as not every environment uses `conda`.

Reply:
I chose to go with `source activate` since it works both for conda and venv. any concerns about that? does `conda run` have any advantage?